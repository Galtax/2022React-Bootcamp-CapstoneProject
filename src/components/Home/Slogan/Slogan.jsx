import React from 'react';
import PropTypes from 'prop-types';
import SloganFurny from './SloganFurny.styled';

function Slogan({ text }) {
  return (
    <SloganFurny>
      <h1>{text}</h1>
    </SloganFurny>
  );
}

Slogan.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Slogan;
