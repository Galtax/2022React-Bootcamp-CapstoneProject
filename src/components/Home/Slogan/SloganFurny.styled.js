import styled from 'styled-components';
import { bluePrince } from '../../../utils/constant.styles';

const SloganFurny = styled.div`
  margin-top: 0;
  background: rgba(250,250,250,.4);
  font-size: 2.2rem;
  text-align: center;
  color: ${bluePrince};

  & h1 {
    font-family: 'Tangerine', cursive;
  } 
  
  @media screen and (max-width: 700px) {
    font-size: 1.7rem;
  }
`;

export default SloganFurny;
