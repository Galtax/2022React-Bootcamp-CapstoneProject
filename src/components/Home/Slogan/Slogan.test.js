import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-styled-components';
import Slogan from '.';

describe('Slogan Component Test', () => {

  it('should render the Slogan Component correctly', () => {
    const wrapper = mount(<Slogan text='Lorem Ipsum' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should contain subcomponents', () => {
    const mockText = 'Test slogan text';
    let wrapper = shallow(<Slogan text={mockText} />);
    
    expect(wrapper.exists()).toEqual(true);
    expect(wrapper.text()).toEqual(mockText);
  });
});
