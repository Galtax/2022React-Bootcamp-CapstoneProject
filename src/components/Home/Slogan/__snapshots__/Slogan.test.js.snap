// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slogan Component Test should render the Slogan Component correctly 1`] = `ReactWrapper {}`;
